### PR TITLE
Opening & closing of preview is causing unexpected behavior, if we have multiple Question-sets in the content.

### PR DESCRIPTION
**Pull request by JagadishPujari**
[JIRA LINK](https://project-sunbird.atlassian.net/browse/SB-8095)

### CODE REVIEW

File: org.ekstep.navigation-1.0/renderer/controller/navigation_ctrl.js [hunk 1]
```diff
@@ -144,7 +144,8 @@ angular.module('genie-canvas').directive('customNextNavigation', function ($root
 
     $scope.pluginInstance = EkstepRendererAPI.getPluginObjs("org.ekstep.navigation");
 
-    EventBus.addEventListener("sceneEnter", function (data) {
+    EkstepRendererAPI.addEventListener("sceneEnter", function (data) {
+      $scope.pluginInstance._customNavigationPlugins = [];
```
[the remark this answers is not in the document]
Comment (JagadishPujari) [reply org.ekstep.navigation-1.0/renderer/controller/navigation_ctrl.js]: Done, Added `renderer:stage:reset` event and clearing registered plugin for navigation.

File: org.ekstep.navigation-1.0/renderer/plugin.js [hunk 2]
```diff
@@ -19,6 +19,7 @@
     org.ekstep.service.controller.loadNgModules(this._templatePath, this.controllerPath);
 
     EkstepRendererAPI.addEventListener("renderer:overlay:show", instance.showOrHideOverlay, instance);        
+    EkstepRendererAPI.addEventListener("renderer:stage:reset", instance.resetNavigation, instance);
```
Comment (vinukumar-vs) [inline org.ekstep.navigation-1.0/renderer/plugin.js]: Add this code in the controller. It has below event listeners in the init() method 
EkstepRendererAPI.addEventListener("renderer:content:start", $scope.showCustomNavigation);
    EkstepRendererAPI.addEventListener("renderer:overlay:show", $scope.showCustomNavigation);
    EkstepRendererAPI.addEventListener("renderer:overlay:hide", $scope.hideCustomNavigation);
 
Add EkstepRendererAPI.addEventListener("renderer:stage:reset", instance.resetNavigation, instance); as well

File: org.ekstep.navigation-1.0/renderer/plugin.js [hunk 3]
```diff
@@ -77,6 +78,9 @@
   initPlugin: function (data) {
       // Plugin actions are handled in the angularJS controller.
   },
+  resetNavigation: function(){
```
Comment (vinukumar-vs) [inline org.ekstep.navigation-1.0/renderer/plugin.js]: Move this function to the controller. Both event listener & function should be in the controller.

File: org.ekstep.navigation-1.0/renderer/plugin.js [hunk 4]
```diff
@@ -18,9 +18,6 @@
     this.controllerPath = org.ekstep.pluginframework.pluginManager.resolvePluginResource(this._manifest.id, this._manifest.ver, "renderer/controller/navigation_ctrl.js");
     org.ekstep.service.controller.loadNgModules(this._templatePath, this.controllerPath);
 
-    EkstepRendererAPI.addEventListener("renderer:overlay:show", instance.showOrHideOverlay, instance);        
-    EkstepRendererAPI.addEventListener("renderer:content:start", instance.showOrHideOverlay, instance);
```
Comment (vinukumar-vs) [inline org.ekstep.navigation-1.0/renderer/plugin.js]: Dont remove this code. In controller, it is not updating to plugininstance.customNavigationVisible  variable.

**Pull request merged**